base_report_auto_create_qweb/models/report_xml.py

```diff
@@ -0,0 +1,95 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, api, exceptions, _
+
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report.xml'
+
+    def _prepare_qweb_view_data(self, qweb_name, arch):
+        return {
+            'name': qweb_name,
+            'mode': 'primary',
+            'type': 'qweb',
+            'arch': arch,
+        }
+
+    def _prepare_model_data_data(self, qweb_name, module, qweb_view):
+        return {
+            'module': module,
+            'name': qweb_name,
+            'res_id': qweb_view.id,
+            'model': 'ir.ui.view',
+        }
+
+    def _prepare_value_view_data(self, name, model):
+        return {
+            'name': name,
+            'model': model,
+            'key2': 'client_print_multi',
+            'value_unpickle': 'ir.actions.report.xml,%s' % self.id,
+        }
+
+    def _create_qweb(self, name, qweb_name, module, model, arch):
+        qweb_view_data = self._prepare_qweb_view_data(qweb_name, arch)
+        qweb_view = self.env['ir.ui.view'].create(qweb_view_data)
+        model_data_data = self._prepare_model_data_data(
+            qweb_name, module, qweb_view)
+        self.env['ir.model.data'].create(model_data_data)
+        value_view_data = self._prepare_value_view_data(
+            name, model)
+        self.env['ir.values'].create(value_view_data)
```

Comment:
Tested on runbot... User *Demo* with **Administration/Settings** and **Technical Features** groups
```
Exception during JSON request handling.
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/http.py", line 537, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/http.py", line 574, in dispatch
    result = self._call_function(**self.params)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/http.py", line 310, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/service/model.py", line 113, in wrapper
    return f(dbname, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/http.py", line 307, in checked_call
    return self.endpoint(*a, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/http.py", line 803, in __call__
    return self.method(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/http.py", line 403, in response_wrap
    response = f(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/addons/web/controllers/main.py", line 944, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/addons/web/controllers/main.py", line 936, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 241, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 336, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/addons/base_report_auto_create_qweb/models/report_xml.py", line 76, in create
    report_xml._create_qweb(name, report_name, module, model, arch)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/addons/base_report_auto_create_qweb/models/report_xml.py", line 44, in _create_qweb
    self.env['ir.values'].create(value_view_data)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 239, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 463, in new_api
    result = method(self._model, cr, uid, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/addons/base/ir/ir_values.py", line 193, in create
    res = super(ir_values, self).create(cr, uid, vals, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 241, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 336, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/models.py", line 4077, in create
    record = self.browse(self._create(old_vals))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 239, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 463, in new_api
    result = method(self._model, cr, uid, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/models.py", line 4262, in _create
    result += self._columns[field].set(cr, self, id_new, field, vals[field], user, rel_context) or []
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/osv/fields.py", line 1491, in set
    self._fnct_inv(obj, cr, user, id, name, value, self._fnct_inv_arg, context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/addons/base/ir/ir_values.py", line 126, in _value_pickle
    self.write(cursor, user, id, {name[:-9]: value}, context=ctx)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 241, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/addons/base/ir/ir_values.py", line 198, in write
    res = super(ir_values, self).write(cr, uid, ids, vals, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 241, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 363, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/models.py", line 3777, in write
    self._write(old_vals)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 239, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 547, in new_api
    result = method(self._model, cr, uid, self.ids, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/models.py", line 3882, in _write
    self.check_access_rule(cr, user, ids, 'write', context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 241, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/models.py", line 3529, in check_access_rule
    self._check_record_rules_result_count(cr, uid, sub_ids, returned_ids, operation, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/api.py", line 241, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3112966-195-4e0c1e/openerp/models.py", line 3478, in _check_record_rules_result_count
    (self._description, operation))
except_orm: (u'Access Denied', u'The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: ir.values, Operation: write)')
```